### ISSUE TITLE
PL2 Performance analysis

### ISSUE THREAD

**Issue by bradfrost** [pattern-lab/edition-php-development#12]
![webpagetest test details - dulles dev patternlab io - 04 30 15 13 23 54](https://cloud.githubusercontent.com/assets/383701/7420002/96450e52-ef3e-11e4-8857-cd93f8f62e74.png)

Here's what the waterfall looks like right now. Not sure what the big gap is all about. Locally, PL will hang for a while from time to time, but I don't know what's causing that.

**Comment by bradfrost** [comment]
@dmolsen

**Comment by dmolsen** [comment]
If the "hang" is on the "find a pattern" that was a bad recursive function. Basically I set focus on an element that then set focus on itself which set focus on itself which... You get the idea. We're at ~300k in JS/CSS that we can control. I don't think that's overly bad. I can look but... a second for local dev doesn't sound or feel horrible.

**Comment by bradfrost** [comment]
Good find with the recursive pattern! 

> We're at ~300k in JS/CSS that we can control.

Yep, I'd say that's pretty good.

I'd still like to see us consolidate/clean up some stuff in the footer to keep what we expose to the user to a minimum. Seeing things like `bower_components` might send some people running for the hills. Not saying we can't use that stuff, but it would be easier for folks to rip out when it comes time to migrate into a production environment.

**Comment by dmolsen** [comment]
I think you're confusing the viewer (the iframe container) and the patterns (what's in the iframe). Why would someone need to rip out anything from the viewer? There is one call to `bower_components/` in a pattern. Looking at the pattern it's a little dirty in the footer but I need to be able to include the `postMessage` stuff because of browser security policies.

**Comment by bradfrost** [comment]
> I think you're confusing the viewer (the iframe container) and the patterns (what's in the iframe). Why would someone need to rip out anything from the viewer?

Yep I think you're right about that. People wouldn't really be touching that stuff. That said, I think it's still a good idea to consolidate that stuff to reduce roundtrips to the server. If that mucks everything up, I'm fine leaving it as is!

> Looking at the pattern it's a little dirty in the footer but I need to be able to include the postMessage stuff because of browser security policies.

Ah good to know. Yeah I was wondering what was going on with those inline scripts, but figured they were like that for a reason.

**Comment by dmolsen** [comment]
It doesn't necessarily muck everything up. I just don't think there are big gains to be made. The PL front-end is a grand total of 14 requests before plug-ins are added. As we get closer to moving to production I'll concat some of those bower_components together.
